eclair-core/src/test/scala/fr/acinq/eclair/wire/internal/channel/ChannelCodecsSpec.scala

```diff
@@ -334,4 +334,34 @@ object ChannelCodecsSpec {
     DATA_NORMAL(MetaCommitments(commitments), ShortIds(RealScidStatus.Final(RealShortChannelId(42)), ShortChannelId.generateLocalAlias(), None), None, channelUpdate, None, None, None)
   }
 
+  def makeChannelDataPending(remoteNodeId: Crypto.PublicKey, channelId: ByteVector32): DATA_WAIT_FOR_CHANNEL_READY = {
+    val fundingTx = Transaction.read("0200000001adbb20ea41a8423ea937e76e8151636bf6093b70eaff942930d20576600521fd000000006b48304502210090587b6201e166ad6af0227d3036a9454223d49a1f11839c1a362184340ef0240220577f7cd5cca78719405cbf1de7414ac027f0239ef6e214c90fcaab0454d84b3b012103535b32d5eb0a6ed0982a0479bbadc9868d9836f6ba94dd5a63be16d875069184ffffffff028096980000000000220020c015c4a6be010e21657068fc2e6a9d02b27ebe4d490a25846f7237f104d1a3cd20256d29010000001600143ca33c2e4446f4a305f23c80df8ad1afdcf652f900000000")
+    val fundingAmount = fundingTx.txOut.head.amount
+    val commitmentInput = Funding.makeFundingInputInfo(fundingTx.hash, 0, fundingAmount, channelKeyManager.fundingPublicKey(localParams.fundingKeyPath).publicKey, remoteParams.fundingPubKey)
+
+    val remoteSig = ByteVector64(hex"2148d2d4aac8c793eb82d31bcf22d4db707b9fd7eee1b89b4b1444c9e19ab7172bab8c3d997d29163fa0cb255c75afb8ade13617ad1350c1515e9be4a222a04d")
+    val commitTx = Transaction(
+      version = 2,
+      txIn = TxIn(
+        outPoint = commitmentInput.outPoint,
+        signatureScript = ByteVector.empty,
+        sequence = 0,
+        witness = Scripts.witness2of2(randomBytes64(), remoteSig, channelKeyManager.fundingPublicKey(localParams.fundingKeyPath).publicKey, remoteParams.fundingPubKey)) :: Nil,
+      txOut = Nil,
+      lockTime = 0
+    )
+    val localCommit = LocalCommit(0, CommitmentSpec(Set(), FeeratePerKw(1500 sat), 50000000 msat, 70000000 msat), CommitTxAndRemoteSig(CommitTx(commitmentInput, commitTx), remoteSig), Nil)
+    val remoteCommit = RemoteCommit(0, CommitmentSpec(Set(), FeeratePerKw(1500 sat), 50000 msat, 700000 msat), ByteVector32(hex"0303030303030303030303030303030303030303030303030303030303030303"), PrivateKey(ByteVector.fill(32)(4)).publicKey)
+    val commitments = Commitments(channelId, ChannelConfig.standard, ChannelFeatures(), localParams, remoteParams.copy(nodeId = remoteNodeId), ChannelFlags.Public, localCommit, remoteCommit, LocalChanges(Nil, Nil, Nil), RemoteChanges(Nil, Nil, Nil),
+      localNextHtlcId = 32L,
+      remoteNextHtlcId = 4L,
+      originChannels = Map(),
+      remoteNextCommitInfo = Right(randomKey().publicKey),
+      localFundingStatus = UnknownFundingTx,
+      remoteFundingStatus = RemoteFundingStatus.NotLocked,
+      remotePerCommitmentSecrets = ShaChain.init)
+
+    DATA_WAIT_FOR_CHANNEL_READY(MetaCommitments(commitments), ShortIds(RealScidStatus.Final(RealShortChannelId(42)), ShortChannelId.generateLocalAlias(), None))
+  }
```

Comment:
I don't think this is necessary, you should be able to just use `CommitmentsSpec.makeCommitments` to get a `MetaCommitments` object, and from this it should be trivial to create some `DATA_WAIT_FOR_FUNDING_CONFIRMED`, `DATA_WAIT_FOR_CHANNEL_READY`, `DATA_WAIT_FOR_DUAL_FUNDING_CONFIRMED` and `DATA_WAIT_FOR_DUAL_FUNDING_READY` for test cases.

You only care about setting the `remoteNodeId` and reading the `channelId` in your tests, right?

Reply:
I rolled back my changes to `ChannelCodecsSpec.scala` and replaced it with a simpler `makeChannelDataPending`  function local to the tests that use it and create the metaCommitments as you suggested. See [b4c3ba0](https://github.com/ACINQ/eclair/pull/2552/commits/b4c3ba0146aa599d3642d417e18c4d1547c41179).

Do you think I should create tests for each of the four pending states that are filtered for when you restore pending requests from the channel db? Right now I use `DATA_WAIT_FOR_DUAL_FUNDING_READY` as a proxy for the others since the code is so simple there.